lib/gourami/attributes.rb

```diff
@@ -23,39 +23,42 @@ def attribute(name, options = {}, &default_block)
         options = options.dup
         options[:default] = default_block if block_given?
 
-        unless options[:skip_reader]
-          define_method(:"#{name}") do
-            value = instance_variable_get(:"@#{name}")
-            default = options[:default]
-
-            if value.nil? && default
-              default.respond_to?(:call) ? instance_exec(&default) : default
-            else
-              value
+        mixin = Module.new do |mixin|
+          unless options[:skip_reader]
+            mixin.send(:define_method, :"#{name}") do
+              value = instance_variable_get(:"@#{name}")
+              default = options[:default]
+
+              if value.nil? && default
+                default.respond_to?(:call) ? instance_exec(&default) : default
+              else
+                value
+              end
             end
           end
-        end
 
-        # Define external setter.
-        define_method(:"#{name}=") do |value|
-          provided_attributes_names[name.to_s] = options
-          send(:"_#{name}=", value)
-        end
+          # Define external setter.
+          mixin.send(:define_method, :"#{name}=") do |value|
+            provided_attributes_names[name.to_s] = options
+            send(:"_#{name}=", value)
+          end
 
-        # Define internal setter.
-        define_method(:"_#{name}=") do |value|
-          value = setter_filter(name, value, options)
-          instance_variable_set(:"@#{name}", value)
-        end
-        private :"_#{name}="
+          # Define internal setter.
+          mixin.send(:define_method, :"_#{name}=") do |value|
+            instance_variable_set(:"@#{name}", setter_filter(name, value, options))
+          end
+          mixin.send(:private, :"_#{name}=")
 
-        case options[:type]
-        when :boolean
-          define_method(:"#{name}?") do
-            !!send(name)
+          case options[:type]
+          when :boolean
+            mixin.send(:define_method, :"#{name}?") do
+              !!send(name)
+            end
```

Comment:
do you need a case statement to just check if boolean?

Reply:
I mean not really but leaves open for more